crates/astria-proto/src/native/sequencer/v1alpha1.rs

```diff
@@ -1251,6 +1266,1158 @@ impl Display for IncorrectAddressLength {
+    // sequencer block is derived form. This proof together with
+    // `Sha256(MTH(rollup_transactions))` must match `header.data_hash`.
+    // `MTH(rollup_transactions)` is the Merkle Tree Hash derived from the
+    // rollup transactions.
+    rollup_transactions_proof: merkle::Proof,
+    // The proof that the rollup IDs listed in `rollup_transactions` are included
+    // in the `CometBFT` block this sequencer block is derived form. This proof together
+    // with `Sha256(MTH(rollup_ids))` must match `header.data_hash`.
+    // `MTH(rollup_ids)` is the Merkle Tree Hash derived from the rollup IDs listed in
+    // the rollup transactions.
+    rollup_ids_proof: merkle::Proof,
+}
+
+impl SequencerBlock {
+    /// Returns the hash of the `CometBFT` block this sequencer block is derived from.
+    ///
+    /// This is done by hashing the `CometBFT` header stored in this block.
+    #[must_use]
+    pub fn block_hash(&self) -> [u8; 32] {
+        self.block_hash
+    }
+
+    #[must_use]
+    pub fn header(&self) -> &tendermint::block::header::Header {
+        &self.header
+    }
+
+    #[must_use]
+    pub fn rollup_transactions(&self) -> &IndexMap<RollupId, Vec<Vec<u8>>> {
+        &self.rollup_transactions
+    }
+
+    #[must_use]
+    pub fn into_raw(self) -> raw::SequencerBlock {
+        fn tuple_to_rollup_txs(
+            (rollup_id, transactions): (RollupId, Vec<Vec<u8>>),
+        ) -> raw::RollupTransactions {
+            raw::RollupTransactions {
+                id: rollup_id.to_vec(),
+                transactions,
+            }
+        }
+
+        let Self {
+            header,
+            rollup_transactions,
+            rollup_transactions_proof,
+            rollup_ids_proof,
+            ..
+        } = self;
+        raw::SequencerBlock {
+            header: Some(header.into()),
+            rollup_transactions: rollup_transactions
+                .into_iter()
+                .map(tuple_to_rollup_txs)
+                .collect(),
+            rollup_transactions_proof: Some(rollup_transactions_proof.into_raw()),
+            rollup_ids_proof: Some(rollup_ids_proof.into_raw()),
+        }
+    }
+
+    #[must_use]
+    pub fn into_unchecked(self) -> UncheckedSequencerBlock {
+        let Self {
+            header,
+            rollup_transactions,
+            rollup_transactions_proof,
+            rollup_ids_proof,
+            ..
+        } = self;
+        UncheckedSequencerBlock {
+            header,
+            rollup_transactions,
+            rollup_transactions_proof,
+            rollup_ids_proof,
+        }
+    }
+
+    /// Turn the sequencer block into a [`CelestiaSequencerBlob`] and its associated list of
+    /// [`CelestiaRollupBlob`]s.
+    #[must_use]
+    #[allow(clippy::missing_panics_doc)] // the proofs are guaranteed to exist; revisit if refactoring
+    pub fn to_celestia_blobs(&self) -> (CelestiaSequencerBlob, Vec<CelestiaRollupBlob>) {
```

Comment:
This method used to be a free standing method in `astria-celestia-client`. Because all these types are defined here it's natural to leave this here.